cpp/CMakeLists.txt

```diff
@@ -1,6 +1,11 @@
 cmake_minimum_required (VERSION 2.8)
```

Comment:
Yeah I agree, as the purpose of agents is completely different from gym_tcp_api, it would be better  to be moved to the models repo. :+1:
The Gym related files should also be moved
```
> Deps
   > Gym_cpp
> Models
   > Agents
       > Agent 1
       > Agent 2
       ....
```
Will this folder structure work fine?